### PR TITLE
IHandleSagaNotFound is async

### PR DESCRIPTION
**Pull request by danielmarbach**
This PR is part of the work to make the core async enabled. For more background read [Async/Await: It's time!](http://particular.net/blog/async-await-its-time) blogpost

IHandleSagaNotFound is async

@Particular/tf-asyncawait please review

### CODE REVIEW

**Comment by danielmarbach** [comment]
@SimonCropp rebase was super smooth :dancer:

**Comment by SimonCropp** [comment]
@danielmarbach cool. i was expecting the inevitable "damn simon chainging the same code"

File: src/NServiceBus.AcceptanceTests/Sagas/When_timeout_hit_not_found_saga.cs [hunk 1]
```diff
@@ -83,7 +83,7 @@ public void Handle(object message)
                     {
                         Context.NotFoundHandlerCalledForTimeout = true;
                     }
-
+                    return Task.FromResult(0);
```
Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Sagas/When_timeout_hit_not_found_saga.cs]: Use the new TaskEx?

Comment (SimonCropp) [inline src/NServiceBus.AcceptanceTests/Sagas/When_timeout_hit_not_found_saga.cs]: nope. plan is to update acceptance tests to 4.6. then no need to make taskex public

Comment (SimonCropp) [inline src/NServiceBus.AcceptanceTests/Sagas/When_timeout_hit_not_found_saga.cs]: but will do the 4.6 thing after @distantcam pulls the trigger

**Comment by danielmarbach** [comment]
@SimonCropp or @andreasohlund can you merge this?

**Pull request merged**